cheroot/test/test_https.py

```diff
@@ -0,0 +1,118 @@
+"""HTTPS Tests."""
+# -*- coding: utf-8 -*-
+# vim: set fileencoding=utf-8 :
+
+import os
+import pytest
+import ssl
+
+from cheroot.ssl.builtin import BuiltinSSLAdapter
+from cheroot.test import helper
+import cheroot
+import ddt
+
+
+def create_wsgi_server(**conf):
+    ssl_adapter = BuiltinSSLAdapter(
+        certificate=conf.pop('certificate'),
+        private_key=conf.pop('private_key'),
+        certificate_chain=conf.pop('certificate_chain'),
+        verify_mode=conf.pop('verify_mode', None))
+    server = cheroot.wsgi.Server(**conf)
+    server.ssl_adapter = ssl_adapter
+    return server
+
+
+def ssl_file(filename):
+    return os.path.join('cheroot/test/ssl/', filename)
+
+
+class HTTPSTestBase(object):
+
+    available_servers = {'wsgi': create_wsgi_server}
+    config = {
+        'certificate': 'cheroot/test/ssl/server.cert',
+        'private_key': 'cheroot/test/ssl/server.key',
+        'certificate_chain': 'cheroot/test/ssl/ca.cert',
+        'verify_mode': None
+    }
+    script_name = '/hello'
+    ssl_context = None
+
+    @classmethod
+    def setup_server(cls):
+        class Root(helper.Controller):
+
+            def hello(self, req, resp):
+                return 'Hello world!'
+
+        cls.httpserver.wsgi_app = Root()
+
+    def setUp(self):
+        super().setUp()
+        self.ssl_context = ssl.create_default_context(cafile='cheroot/test/ssl/ca.cert')
+        self.ssl_context.check_hostname = False
+
+    def getPage(self, *args, **kw):
+        return super().getPage(self.script_name, *args, raise_subcls=ssl.SSLError, **kw)
+
+    def set_client_cert(self, name):
+        self.ssl_context.load_cert_chain(ssl_file(name + '.cert'), keyfile=ssl_file('client.key'))
+
+    def assert_allowed(self, client_cert):
+        self.set_client_cert(client_cert)
+        self.getPage()
+        self.assertStatus('200 OK')
+
+    def assert_reject(self, client_cert):
+        self.set_client_cert(client_cert)
+        with pytest.raises(ssl.SSLError) as context:
+            self.getPage()
+        return context
+
+
+@ddt.ddt
```

Comment:
I like the use of ddt, and if weren't married to unittest by way of webtest and could use pytest fixtures instead, I'd prefer those. But I'm glad to see these work here.

Reply:
And we fully support using pytest fixtures now :)

Reply:
#67 (FTR)